api/src/repositories/survey-repository.ts

```diff
@@ -739,6 +739,63 @@ export class SurveyRepository extends BaseRepository {
     return result.id;
   }
 
+  /**
+   * Insert a new row for each provided intended outcome id.
+   *
+   * @param {number[]} intendedOutcomeIds
+   * @param {number} surveyId
+   */
+  async insertSurveyIntendedOutcomes(intendedOutcomeIds: number[], surveyId: number): Promise<void> {
+    const knex = getKnex();
+    const queryBuilder = knex.queryBuilder();
+    const rowsForInsertion = intendedOutcomeIds.map((outcome_id) => ({
+      intended_outcome_id: outcome_id,
+      survey_id: surveyId
+    }));
+    queryBuilder.insert(rowsForInsertion).into('survey_intended_outcome');
+    await this.connection.knex(queryBuilder);
+    return;
+  }
+
+  /**
+   * Insert new rows that weren't found in the existing list. Delete rows that existed before but weren't found in the new list.
+   * @param {number[]} intendedOutcomeIds
+   * @param {number[]} existingOutcomeIds
+   * @param {number} surveyId
+   */
+  async updateSurveyIntendedOutcomes(
```

Comment:
I think the decision making aspect of this code should live in the survey service. We try and keep the repos as simple as we can. A good example of this pattern is in the `insertUpdateDeleteSurveyLocation` function in the survey service